{
  try {
    return method.invoke(thiz,args);
  }
 catch (  Exception e) {
    throw new RuntimeException("failed to invoke " + method + ": "+ e,e);
  }
}
